src/linear_algebra/finsupp.lean

```diff
@@ -632,3 +633,16 @@ lemma mem_span_finset {s : finset M} {x : M} :
     (show x ∈ span R (id '' (↑s : set M)), by rwa set.image_id) in
   ⟨v, hvx ▸ (finsupp.total_apply_of_mem_supported _ hvs).symm⟩,
 λ ⟨f, hf⟩, hf ▸ sum_mem _ (λ i hi, smul_mem _ _ $ subset_span hi)⟩
+
+/-- An element `m ∈ M` is contained in the `R`-submodule spanned by a set `s ⊆ M`, if and only if
+`m` can be written as a finite `R`-linear combination of elements of `s`.
+The implementation uses `finsupp.sum`. -/
+lemma mem_span_set {m : M} {s : set M} :
+  m ∈ submodule.span R s ↔
+  ∃ c : M →₀ R, (c.support : set M) ⊆ s ∧ c.sum (λ mi r, r • mi) = m :=
+begin
+  convert @finsupp.mem_span_iff_total _ M R _ _ _ id s m,
+  { exact (set.image_id s).symm },
+  { ext,
+    split; exact λ ⟨hsub, hsum⟩, ⟨hsub, hsum⟩ }
```

Comment:
```suggestion
  { ext,
    rw exists_prop,
    refl,⟩ }
```

Reply:
I think that this suggestion is now outdated.